pkg/ipamd/ipamd.go

```diff
@@ -1201,15 +1196,12 @@ func (c *IPAMContext) askForTrunkENIIfNeeded(ctx context.Context) {
 	if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
 		// Check that there is room for a trunk ENI to be attached:
 		if c.dataStore.GetENIs() >= (c.maxENI - c.unmanagedENI) {
-			log.Debug("No slot available for a trunk ENI to be attached. Not labeling the node")
+			log.Debug("No slot available for a trunk ENI to be attached. Will not create a node event for resource controller")
 			return
 		}
 		// We need to signal that VPC Resource Controller needs to attach a trunk ENI
-		err := c.SetNodeLabel(ctx, "vpc.amazonaws.com/has-trunk-attached", "false")
-		if err != nil {
-			podENIErrInc("askForTrunkENIIfNeeded")
-			log.Errorf("Failed to set node label", err)
-		}
+		log.Debug("VPC CNI is asking RC to initialize trunk interface")
+		c.eventRecorder.SendNodeEvent(corev1.EventTypeNormal, eventrecorder.EventReason, "NeedTrunk", "vpc.amazonaws.com/has-trunk-attached=false")
```

Comment:
If node does not support `has-trunk-attached`, can we avoid sending this event and log error for readability?

Reply:
It is a bit harder for now because we haven't verified if we can validate node type for the feature during runtime. But this is planned to be done for the next release of VPC resource controller if APIs are available from EC2. You can refer to t[his open issue](https://github.com/aws/amazon-vpc-resource-controller-k8s/issues/98).
Otherwise we have to maintain a hard copy of file listing all supported types, which is exactly what we want to deprecate.